### PR TITLE
serial_msp test code

### PR DESCRIPTION
**Pull request by martinbudden**
Test code for the *Multiwii Serial Protocol* (MSP), `serial_msp.c`

Processes some simple *MSP* commands (`MSP_API_VERSION`, `MSP_FC_VARIANT`, `MSP_FC_VERSION`, `MSP_PID_CONTROLLER`) and checks they are written out correctly.

Round trips the `MSP_PID` command: writes out typical PID data using `MSP_PID`and reads it back in again using `MSP_SET_PID` and checks that the read data is the same as the writtend date. Does this for both integer and floating point PID.

For minimal intrusion, the test code duplicates some of the MSP command `#defines`, the `mspState_e` `enum` and the  `mspPort_t` structure. These should probably be moved out of `msp_serial.c` into a header file.

### CODE REVIEW

File: src/test/unit/serial_msp_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+#define MSP_FC_VERSION                  3    //out message
+
+#define MSP_PID_CONTROLLER              59
+#define MSP_SET_PID_CONTROLLER          60
+
+#define MSP_PID                  112    //out message         P I D coeff (9 are used currently)
+#define MSP_PIDNAMES             117    //out message         the PID names
+#define MSP_SET_PID              202    //in message          P I D coeff (9 are used currently)
+
+#define INBUF_SIZE 64
```
Comment (hydra) [inline src/test/unit/serial_msp_unittest.cc]: prefix each of these `#defines` with `TEST_`.  This will increase test robustness and be less confusing.

**Comment by hydra** [comment]
@martinbudden It's great to see these tests.  Thanks for taking the time to write them.  Much appreciated!

**Comment by hydra** [comment]
With regards to moving `mspState_e` enum and the `mspPort_t` structure to a new file.  Feel free to do this.

**Comment by martinbudden** [comment]
Deleted and recreated serial_msp_unittest branch

**Pull request closed without merging**